jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/HttpStream.java

```diff
@@ -119,12 +124,14 @@ default Throwable consumeAvailable()
             content.release();
 
             // if the input failed, then fail the stream for same reason
-            if (content instanceof Content.Chunk.Error error)
+            if (content instanceof Chunk.Error error)
                 return error.getCause();
 
             if (content.isLast())
                 return null;
         }
+
+        return new IOException("Content not consumed");
```

Comment:
This exception and the one few lines above should be `StaticException` instances for performance reasons.

Reply:
done